src/main/java/io/orangebeard/listener/OrangebeardExtension.java

```diff
@@ -72,20 +76,68 @@ public OrangebeardExtension() {
         startTestRunAndAddShutdownHook(orangebeardProperties);
     }
 
+    /**
+     * Given an ExtensionContext, determine the fully qualified name (canonical name) of the class that it refers to.
+     * If the ExtensionContext does not have a class (as can happen in a unit test), then it returns an empty String.
+     * @param extensionContext ExtensionContext for a class that is being tested.
+     * @return The fully qualified name of the class that is being tested, or an empty String if there is no such class.
+     */
+    private String getFullyQualifiedClassName(ExtensionContext extensionContext) {
```

Comment:
Private methods should (generally) be below all the other public methods. 

Furthermore, the only thing you require from the extensioncontext is the Class. So I think that should be the parameter of this method.